### PR TITLE
Bluetooth: Implement Secondary_Advertising_Max_Skip in controller

### PR DESCRIPTION
**Pull request by Tronil**
Fix the host side not setting sec_adv_max_skip

Behaviour is unchanged for max_skip == 0

For max_skip > 0, use a slightly different calculation to ensure we can pass LL/DDI/ADV/BV-28-C

### CODE REVIEW

File: subsys/bluetooth/controller/Kconfig.ll_sw_split [hunk 1]
```diff
@@ -335,6 +335,15 @@ config BT_CTLR_ADV_AUX_SET
...
 
+config BT_CTLR_ADV_USE_MAX_SECONDARY_INTERVAL
+	bool "Use the maximum possible interval between ADV_AUX_INDs"
+	depends on BT_BROADCASTER && BT_CTLR_ADV_EXT
+	default n
+	help
+	  Use the value of Secondary_Advertising_Max_Skip to only send
+	  ADV_AUX_IND when required - ie. skip as many as allowed by the
+	  parameters set by the host.
+
```
Comment (cvinayak) [inline subsys/bluetooth/controller/Kconfig.ll_sw_split:346]: If this is not something configurable in IXIT or ICS, lets not have a Kconfig (in this PR Host will always provide a valid value anyway).

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Comment by jori-nordic** [comment]
> Fix the host side not setting sec_adv_max_skip

Do we have a test for that?

File: subsys/bluetooth/controller/Kconfig.ll_sw_split (hunk 1 above)
Comment (Tronil) [reply subsys/bluetooth/controller/Kconfig.ll_sw_split:346]: True, that would be preferable; However, this has some implications for qualification (specifically LL/DDI/ADV/BV-28-C). I'll see if I can get some time to look at it next week, but otherwise it will have to wait until I get back from my summer vacation (ie. late August).

**Comment by Thalley** [comment]
> > Fix the host side not setting sec_adv_max_skip
> 
> Do we have a test for that?

Clearly not, since we've never set the value. Not sure if that counts as a bug or a missing enhancement :D

File: subsys/bluetooth/host/adv.c [hunk 2]
```diff
@@ -1216,6 +1216,8 @@ static int le_ext_adv_param_set(struct bt_le_ext_adv *adv,
 
 	cp->sid = param->sid;
 
+	cp->sec_adv_max_skip = param->secondary_max_skip;
+
```
Comment (Thalley) [inline subsys/bluetooth/host/adv.c:1220]: Suggest to put this change into a separate commit

File: subsys/bluetooth/controller/Kconfig.ll_sw_split (hunk 1 above)
Comment (Tronil) [reply subsys/bluetooth/controller/Kconfig.ll_sw_split]: Changed the implementation to avoid the configuration flag; This means using a slightly different calculation for max_skip == 0 to avoid a change in behaviour while still passing LL/DDI/ADV/BV-28-C for higher values of max_skip

File: subsys/bluetooth/host/adv.c (hunk 2 above)
Comment (Tronil) [reply subsys/bluetooth/host/adv.c:1220]: Done

File: subsys/bluetooth/controller/ll_sw/ull_adv.c [hunk 3]
```diff
@@ -1487,20 +1488,31 @@ uint8_t ll_adv_enable(uint8_t enable)
...
+				aux->interval =
+					DIV_ROUND_UP(((uint64_t)adv->interval *
+							  ADV_INT_UNIT_US) +
+							 HAL_TICKER_TICKS_TO_US(
+								ULL_ADV_RANDOM_DELAY),
+							 PERIODIC_INT_UNIT_US);
+			} else {
+				aux->interval =
+					(((uint64_t)adv->interval * ADV_INT_UNIT_US) *
+					 (adv->max_skip + 1)) / PERIODIC_INT_UNIT_US;
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_adv.c:1514]: Suggest to put `(uint64_t)adv->interval * ADV_INT_UNIT_US)` into a variable to make this more readable. The variable can be used both places. 

Similarly the `/ PERIODIC_INT_UNIT_US` should be possible to have common code for too.

**Review by Thalley — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]
